### PR TITLE
[io] Remove non-user related doc from TFile class. [skip-ci]

### PR DESCRIPTION
**Pull request by Axel-Naumann**
Also preparing for new IO group doc

### CODE REVIEW

File: io/io/src/TFile.cxx
Comment (eguiraud) [inline io/io/src/TFile.cxx:17]:
```suggestion
A ROOT file is a file-system storing objects, possibly inside directories.
```

**Review by eguiraud — APPROVED** [review]

**Review by bellenot — APPROVED** [review]
I trust you 😉

File: io/io/src/TFile.cxx
Comment (pcanal) [inline io/io/src/TFile.cxx:28]: This information is "essential" developer and advancer user information and belong in the repository (as it can and has evolved).  There should be an explicit link or reference to where in the repository that information lives.

**Review by pcanal — CHANGES_REQUESTED** [review]

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [reply io/io/src/TFile.cxx:28]: That's referenced as `\sa \ref rootio`, see https://root.cern/doc/master/rootio.html

If it's not in there I will add it there; let me know where you want it.

**Review by couet — APPROVED** [review]

File: io/io/src/TFile.cxx
Comment (pcanal) [reply io/io/src/TFile.cxx:28]: That's a bit obscure, especially when 'just' reading the source from vi.  It probably should be clarified that the information is available/recorded(?) in `./io/doc/TFile` and in particular `tfile.md` and `tdirectory.md` 
( as a side note the 'format' in those file, both in the text version and the render version seems (to me) much harder to read than the one here (probably the missing/removed column delimiter)

File: io/io/src/TFile.cxx
Comment (eguiraud) [reply io/io/src/TFile.cxx:17]: Better?

**Comment by bellenot** [comment]
BTW, the table looks much better in the [reference guide](https://root.cern.ch/doc/master/classTFile.html) than the [new one](https://root.cern.ch/doc/master/md_io_doc_TFile_header.html#header).

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [reply io/io/src/TFile.cxx:17]: I wanted to clarify the important point that a `TFile` will end up on your disk, as a regular file. I don't think your current wording gets that across: it talks about the *content* of a `TFile`.

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [reply io/io/src/TFile.cxx:28]: > as a side note the 'format' in those file, both in the text version and the render version seems (to me) much harder to read than the one here (probably the missing/removed column delimiter

Fine, but that shouldn't block this PR.

> That's a bit obscure, especially when 'just' reading the source from vi.

Are you complaining about doxygen syntax being obscure, or you not knowing what's in these files? (Serious question.) Either way, my mission is to deconfuse users not necessarily Philippe, and this PR seems to be a step in the right direction ;-) There are several parts missing, e.g. `TFile::Open` instead of constructor etc - but that info wasn't there to begin with. So - what do we do?

File: io/io/src/TFile.cxx
Comment (eguiraud) [reply io/io/src/TFile.cxx:17]: ooooh "a file-system file" means "it's a file in your file system", not "it's a file the content of which logically works like a file system"?

"An on-disk file" then?

File: io/io/src/TFile.cxx
Comment (pcanal) [reply io/io/src/TFile.cxx:28]: > Fine, but that shouldn't block this PR.

In my opinion it was essential.  This documentation is not just for me but also essential for future maintainer and for external implementer of the format.  It could not be removed if there was not a (readable) alternative available.  Glad-fully that was resolved in a separate PR.

> Are you complaining about doxygen syntax being obscure, or you not knowing what's in these files? (Serious question.)

I am just stating that (possibly due to my missing in-brain doxygen parser) given those strings, I can not find the text file that contains the information.  So maybe we can __add__ outside the view of doxygen the actual name of the file or directory that contains the information (to help with browsing the code through a text editor)

File: io/io/src/TFile.cxx
Comment (Axel-Naumann) [reply io/io/src/TFile.cxx:17]: Sounds great!

**Comment by eguiraud** [comment]
Hi, gentle reminder that we need this patch to fix the repeated text at https://root.cern/doc/master/group__IO.html :smile:

**Review by pcanal — APPROVED** [review]

**Review by pcanal — CHANGES_REQUESTED** [review]
We are still missing a more 'human' readable reminder that the detail of the file format are in io/doc/TFile

File: io/io/src/TFile.cxx
Comment (ferdymercury) [reply io/io/src/TFile.cxx:28]: One idea that might be a good compromise. In order to not confuse users with too much technical information, but at the same time not removing all the info from the source code, one could use a dropdown menu, that would only be shown if one really wants to see more advanced info.
See https://gist.github.com/citrusui/07978f14b11adada364ff901e27c7f61

Comment (ferdymercury) [reply io/io/src/TFile.cxx:28]: Related to my last comment: https://github.com/doxygen/doxygen/pull/9331

Comment (pcanal) [reply io/io/src/TFile.cxx:28]: When do they tag a release with that feature?

Comment (ferdymercury) [reply io/io/src/TFile.cxx:28]: The ROOT online-docs are created with a self-built version of doxygen. If I remember correctly, @couet has compiled doxygen versions from git-master that were not exactly at a specific doxygen release, so maybe he can just 'git pull' and recompile on the build machine.
The next stable release will be 1.9.5 probably in four or five months I guess.

Comment (couet) [reply io/io/src/TFile.cxx:28]: Yes I tested 1.9.5 on my machine first before updating the one on the build machine.
It seems stable I will soon update the build machine.
And yes, we are using the most recent version of doxygen.

Comment (couet) [reply io/io/src/TFile.cxx:28]: Done:

```
sftnight@root-ubuntu-2004-3:~/doxygen/build$ doxygen --version
1.9.5 (fdd8e8e750e1b8e58b4d9cc758293f532278dcca)

```

**Comment by ferdymercury** [comment]
I proposed this alternative: https://github.com/root-project/root/pull/14929
(So this one can be closed.)

**Pull request closed without merging**
